src/ChineseNumbers/StringUtils.cpp

```diff
@@ -0,0 +1,48 @@
+
+#include "StringUtils.h"
+
+#include <algorithm>
+#include <sstream>
+
+std::string ChineseNumbers::StringUtils::TrimZerosAtStart(
+    const std::string& input) {
+  std::stringstream output;
+  bool nonZeroFound = false;
+  for (char it : input) {
+    if (nonZeroFound) {
+      output << it;
+      continue;
+    }
+    if (it != '0') {
+      nonZeroFound = true;
+      output << it;
+      continue;
+    }
+  }
+  return output.str();
+}
+
+std::string ChineseNumbers::StringUtils::TrimZerosAtEnd(
+    const std::string& input) {
+  std::string reversed(input);
+  reverse(reversed.begin(), reversed.end());
+  std::string trimmed = StringUtils::TrimZerosAtStart(reversed);
+  reverse(trimmed.begin(), trimmed.end());
+  return trimmed;
+}
+
+std::string ChineseNumbers::StringUtils::LeftPadding(std::string input,
```

Comment:
Change the type of `input` to `const std::string& input`. Line 47 returns a copy anyway.

Reply:
Fixed in 92e46601d055fb57699128d5d2cb8634a1fef339